website/docs/migration/1.0.0_to_2.0.0.mdx

```diff
@@ -0,0 +1,246 @@
+---
+title: ^1.0.0 to ^2.0.0
+---
+
+import buildIinit from "./build_init";
+import familyAndDispose from "./family_and_dispose";
+import fromStateProvider from "./from_state_provider";
+import oldLifecycles from "./old_lifecycles";
+import {AutoSnippet} from "../../src/components/CodeSnippet";
+
+Riverpod went `2.0`! To see the full list of changes, consult the [Changelog].
+
+In this page, we will focus on:
+- How to migrate from the deprecated `StateNotifier` to the new `Notifier` APIs
+- How to handle the new `AsyncValue` behavior and its new properties
+- How to migrate from `ProviderListener` to `ProviderObserver` for logging purposes
+
+
+## Migrating from `StateNotifier`
+
+Riverpod `2.0` introduced the new `Notifier` and `AsyncNotifier` classes.  
+These two are meant to replace the legacy `StateNotifier` (and `ChangeNotifier`), with the following advantages:
+- Conciseness and a better DX overall
+- Codegen support
+- One less [external dependency]
+- Easier distinction between sync and async providers
```

Comment:
Overall these points feel misleading

The new syntax is all about better async support. Aka have a FutureProvider/StreamNotifier with methods

It also:
- naturally offers a ref to notifiers
- use a similar syntax between codegen and non-codegen
- use a similar syntax between sync and async
- move the logic inside providers into notifiers to have all the logic in one place instead of two